script/GameObjects/Entities/Effects/Pulse.lua

```diff
@@ -22,7 +22,7 @@ Preload.Add (function ()
   meshHead = Gen.Primitive.Billboard(-1, -1, 1, 1)
   meshTail = Gen.Primitive.Billboard(-1, -1, 1, 0)
   shaderHead = Cache.Shader('billboard/quad', 'effect/pulsehead')
-  shaderTail = Cache.Shader('billboard/axis', 'effect/pulsetail')
+  shaderTail = Cache.Shader('billboard/axisInvert', 'effect/pulsetail')
```

Comment:
I feel like adding another Axis shader might be wrong here.
I can't quite work out what the code of the shader is doing and wonder if some of the toCam, look and right variables could have better names to understand what  "the axis " is.

But having commented on that, I would prefer to see us stick with one Axis shader that "makes sense" and then it is up to the users of the shader to provide the correct "axis" which in this case is a matrix I think.

We can see that the shader is used by Thruster and Pulse tail and that the shader seems correct for thruster angle. I was then able to correct the pulse tail using that same shader but by passing a "corrected" matrix to it. My change was in Pulse.lua where I added the '-' here. It was a bit of trial and error because I have forgotten all of my £d matrix theory and can't decipher the aforementioned shader code/variable names.

```
function Pulse:refreshMatrix ()
  self.matrix:free()
  self.matrix = Matrix.LookUp(self.pos, -self.dir, Math.OrthoVector(self.dir))
end
```

The change was enough to sort the Z axis out in the final rendering of the tail.

I think this might be exactly the kind of thing @dgavedissian will instantly spot the problem with. So I have mentioned him here in case he can confirm the shader is ok and that the input matrix is a/the solution - without randomly setting a '-' like I did.

But, I still think the Axis shader should be GIVEN an access (therefore Z) to work on without needing any context.